File: challenge/hybris/bin/ext-content/cmsfacades/testsrc/de/hybris/platform/cmsfacades/exception/ValidationError.java
/*
 * [y] hybris Platform
 *
 * Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This software is the confidential and proprietary information of SAP
 * ("Confidential Information"). You shall not disclose such Confidential
 * Information and shall use it only in accordance with the terms of the
 * license agreement you entered into with SAP.
 */
package de.hybris.platform.cmsfacades.exception;

import org.springframework.validation.AbstractBindingResult;


/**
 * Error class to be used for testing purpose only; implements the {@link org.springframework.validation.Errors}
 * interface.
 */
public class ValidationError extends AbstractBindingResult
{
	private static final long serialVersionUID = 2687980944733761368L;

	public ValidationError(final String objectName)
	{
		super(objectName);
	}

	@Override
	public Object getTarget()
	{
		return null;
	}

	@Override
	protected Object getActualFieldValue(final String s)
	{
		return null;
	}

}

File: challenge/hybris/bin/ext-content/smartedit/web/app/common/dtos/ValidationError.ts
/*
 * [y] hybris Platform
 *
 * Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved.
 *
 * This software is the confidential and proprietary information of SAP
 * ("Confidential Information"). You shall not disclose such Confidential
 * Information and shall use it only in accordance with the terms of the
 * license agreement you entered into with SAP.
 */

/**
 * @ngdoc object
 * @name ValidationError.object:ValidationError
 * @description
 * An object representing the backend response for any erroy of type "ValidationError"
 */
export interface ValidationError {
	language?: string;
	message: string;
	reason: string;
	subject: string;
	subjectType: string;
	errorCode: string;
	type: "ValidationError";
}